### PR TITLE
RSDK-4196 Add support for flat tensors in mlmodel

### PR DESCRIPTION
**Pull request by acmorrow**
Updates the C++ SDK to support the changes from https://github.com/viamrobotics/api/pull/279.

### CODE REVIEW

File: src/viam/sdk/services/mlmodel/client.cpp [hunk 1]
```diff
@@ -23,10 +23,7 @@ namespace sdk {
 
 namespace {
 
-struct tensor_storage_and_views {
-    mlmodel_details::tensor_storage storage;
-    MLModelService::named_tensor_views views;
-};
+constexpr bool kUseFlatTensors = true;
```
Comment (acmorrow) [inline src/viam/sdk/services/mlmodel/client.cpp:26]: I've opted to leave the existing struct based implementation in place so I can easily A/B test. All code selected by `kUseFlatTensors = false` would be removable if we decided to make a breaking change.

File: src/viam/sdk/services/mlmodel/private/proto.cpp [hunk 2]
```diff
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <memory>
```
Comment (acmorrow) [inline src/viam/sdk/services/mlmodel/private/proto.cpp:15]: I've moved these down where they belong.

File: src/viam/sdk/services/mlmodel/private/proto.cpp [hunk 3]
```diff
@@ -338,6 +341,145 @@ class tensor_to_pb_value_visitor : public boost::static_visitor<::grpc::Status>
...
+
+template <typename T>
+MLModelService::tensor_views make_sdk_tensor_from_api_tensor_t(const T* data,
+                                                               std::size_t size,
+                                                               std::vector<std::size_t>&& shape,
+                                                               tensor_storage* ts) {
+    if (!data || (size == 0) || shape.empty()) {
+        std::ostringstream message;
+        message << "Empty or zero length data or shape";
+        throw std::invalid_argument(message.str());
```
Comment (stuqdog) [inline src/viam/sdk/services/mlmodel/private/proto.cpp:438]: (q) just for my own clarity, what is `message` doing here? Naively it looks like we're adding a string and then immediately unpacking that string.

**Review by stuqdog — APPROVED** [review]
LGTM!

File: src/viam/sdk/services/mlmodel/private/proto.cpp (hunk 3 above)
Comment (acmorrow) [reply src/viam/sdk/services/mlmodel/private/proto.cpp:438]: @stuqdog - Merely copy paste from other places where I was adding some state to the string, but here I'm not. I'll remove it.

**Review by bhaney — APPROVED** [review]
looked like there wasn't much to change, but you also mentioned how you already wrote a lot with flat-tensors in mind. LGTM!

**Pull request merged**